dissect/target/plugins/os/windows/wer.py

```diff
@@ -0,0 +1,163 @@
+from dissect.target.helpers.fsutil import Path
+from dissect.target.helpers.record import DynamicDescriptor, TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+from dissect.target.target import Target
+
+camel_case_pattern = re.compile(r"(\S)([A-Z][a-z]+)")
+camel_case_pattern_2 = re.compile(r"([a-z0-9])([A-Z])")
+camel_case_pattern_3 = re.compile(r"(\w)[.\s](\w)")
+
+
+def _collect_wer_data(wer_file: Path) -> tuple[list[tuple[str, str]], dict[str, str]]:
+    """Parse data from a .wer file."""
+    record_values = {}
+    record_fields = []
+    key = None
+    for line in wer_file.read_text("utf-16").splitlines():
+        if match := re.match(re.compile(r"(.+)=(.+)"), line.rstrip()):
+            record_type = "string"
+            name, value = match.group(1), match.group(2)
+
+            # dynamic entry with key and value on separate lines
+            if "].Name" in name and not key:
+                key = value
+                # set key and continue to get value on the next line
+                continue
+
+            # dynamic entry with key and value on the same line
+            elif "]." in name and not key:
+                category, name = name.split(".", 1)
+                key = f"{category.split('[')[0]}{name}"
+
+            if "EventTime" in name:
+                value = wintimestamp(int(value))
+                record_type = "datetime"
+                key = "ts"
+
+            key = _key_to_snake_case(key if key else name)
+
+            record_values[key] = value
+            record_fields.append((record_type, key))
+            # reset key necessary for dynamic entries and ts
+            key = None
+
+    return record_fields, record_values
+
+
+def _collect_wer_metadata(metadata_xml_file: Path) -> tuple[list[tuple[str, str]], dict[str, str]]:
+    """Parse data from a metadata .xml file linked to a .wer file."""
+    record_fields = []
+    record_values = {}
+    file = metadata_xml_file.read_text("utf-16")
+
+    tree = ElementTree.fromstring(file)
+    for metadata in tree.iter("WERReportMetadata"):
+        for category in metadata:
+            for value in category:
+                if record_value := value.text.strip("\t\n"):
+                    key = _key_to_snake_case(f"{category.tag}{value.tag}")
+                    record_fields.append(("string", key))
+                    record_values[key] = record_value
+
+    return record_fields, record_values
+
+
+def _create_record_descriptor(record_name: str, record_fields: list[tuple[str, str]]) -> TargetRecordDescriptor:
+    record_fields.extend(
+        [
+            ("path", "wer_file_path"),
+            ("path", "metadata_file_path"),
+        ]
+    )
+    return TargetRecordDescriptor(record_name, record_fields)
+
+
+def _key_to_snake_case(key: str) -> str:
+    res = camel_case_pattern.sub(r"\1_\2", key)
+    res = camel_case_pattern_2.sub(r"\1_\2", res)
+    return camel_case_pattern_3.sub(r"\1_\2", res).lower()
```

Comment:
```suggestion
def _key_to_snake_case(key: str) -> str:
    for pattern in camel_case_patterns:
        key = pattern.sub(r"\1_\2", key)
    return key.lower()
```
Would this work? With this of course as well:

```python
camel_case_patterns = [
    re.compile(r"(\S)([A-Z][a-z]+)")
    re.compile(r"([a-z0-9])([A-Z])")
    re.compile(r"(\w)[.\s](\w)")
]
```

Reply:
Yep, more elegant :)